### DISCUSSION TITLE
New Update - 12th July 2024 | AI Code Review Agent

### DISCUSSION THREAD

**Discussion by rohankokkulabito** [gitbito/bitoai#217] [General]
### Introducing code review analytics

- Explore our new [analytics dashboards](https://alpha.bito.ai/home/dashboard?view=overview) for in-depth insights into your code review process.
- These user-friendly reports help you track key metrics such as pull requests reviewed, issues found, lines of code reviewed, and understand individual contributions.

### Provide feedback on code review quality

- For GitLab, we've added support for emoji-based reactions, while for GitHub, you can respond to feedback question(s) given at the end of code review comments.
- This will help your team track the quality of suggestions provided by the AI Code Review Agent through the "Acceptance Rate" statistics available on the [code review analytics dashboard](https://alpha.bito.ai/home/dashboard?view=overview).

### Track your bito AI usage

- Stay on top of how your team is using Bito AI.
- With the [Requests Usage dashboard](https://alpha.bito.ai/home/bito-premium/ai-request-usage), you can easily see how many AI requests your team members and AI Agents are making.
- The [Usage Management dashboard](https://alpha.bito.ai/home/bito-premium/usage-management) allows you to manage resources and control costs efficiently.
- This way, you can ensure that you're getting the most out of Bito AI without overspending.

### Enhanced context understanding with prompt engineering

- Our improved AI prompts now offer superior comprehension of your code context, significantly increasing the accuracy in identifying and addressing code issues.
- With Bito AI, critical problems are less likely to be overlooked, ensuring you receive a thorough and reliable code analysis.

### Enhanced understanding of CSS, SCSS, and Terraform

- Bito AI now analyzes both immediate context and broader structure of code in CSS, SCSS, and Terraform files, including nested configurations and dependencies.
- This enables Bito to provide more accurate suggestions, optimizations, and error detections.
- For CSS and SCSS, Bito AI identifies patterns and offers improvements in style management and consistency.
- For Terraform, it understands and optimizes infrastructure as code, ensuring more efficient and scalable deployments.

### Enhanced presentation of code review comments
- We've made changes to how the AI Code Review Agent presents its feedback to enhance clarity and usability.

**Comment by dasboe** [comment]
## Deploy-Bericht 2026-03-24

### woenenn — Production (`v16.4.0`, Commit `60d97a44`)
- **#1225**: Continuous Recording mit Feature-Flag (deaktiviert), persistente Transkriptions-Queue, Recorder State Machine
- **#1257**: Fehlende Transkriptionen bei gleichzeitigen Interviews behoben, non-blocking manuelle Transkription
- **#1270**: appName in server_log für Beta/Production-Unterscheidung
- **#1280**: SurveyJS Upgrade 2.3.3 → 2.5.11

### woenenn — Beta (Commit `8e4ac711`)
Enthält alle Production-Änderungen plus:
- **#888**: API für Choices aus Datenbank (`_questionLookups`), Filter mit Array-Support, Items ohne Filter immer sichtbar
- **#1270**: KI-Bearbeitung der Transkripte via AWS Bedrock, collectFractions-Kopie, TranscriptionCopyQueue
- **#1282**: collectFractions von isVerbatim entkoppelt
- **#1225**: Segment-Transcription Fix (fehlte auf Production)
- Hardening: Streaming S3 Upload, ffmpeg `-y`, Bedrock Null Safety, Number Validation Fix

### survey — Production + Beta (Commit `a35d747`)
- **#888**: choicesByUrl API Endpoint mit Filter-Support und Array-Werten
- **#1268**: data.js Sync mit woenenn (isVerbatim paneldynamic Fix)
- Number Validation Fix in data.js
